### PR TITLE
HTTP/2 server example

### PR DESCRIPTION
**Pull request by n1haldev**
This adds http2 server example to the website as mentioned in https://github.com/hyperium/hyper/issues/3568

### CODE REVIEW

**Comment by seanmonstar** [comment]
Thanks for improving the documentation! I was thinking, since it's so similar to the "hello world" guide, that perhaps what might be best is to just update that guide. Where it links out to the example, add a few words "(and a similar example using HTTP/2)". What do you think?

**Comment by n1haldev** [comment]
Ya I think that is good. I will work on that.

**Comment by n1haldev** [comment]
@seanmonstar I have made the requested changes to this PR as per your recommendations. Just wanted to know if there is any other change to be implemented or if the PR is ready for merging.
PS : This other PR that I have raised, https://github.com/hyperium/hyper/pull/3702, is also related to the same issue.

File: _stable/server/hello-world.md [hunk 1]
```diff
@@ -126,6 +126,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
 
 To see all the snippets put together, check out the [full example][example]!
 
+This example uses the `http1` module to create a server that speaks HTTP/1.
```
Comment (seanmonstar) [inline _stable/server/hello-world.md:129]: I'd move this to after the "Also, if `service_fn`"... and add a subheading, like `## HTTP/2`.

File: _stable/server/hello-world.md [hunk 2]
```diff
@@ -126,6 +126,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
...
+In case you want to use HTTP/2, you can use the `http2` module with a few changes to the
+http1 server. The http2 builder requires an executor to run. Executor should implement the `hyper::rt::Executor` trait.
+
+```rust
+impl<F> hyper::rt::Executor<F> for TokioExecutor
```
Comment (seanmonstar) [inline _stable/server/hello-world.md:134]: Instead of this code example inline, you can link to the `../init/runtime.md` guide.

**Review by seanmonstar — COMMENTED** [review]
Thanks for reminding me about the example. I'm glad you're helping make it easier for people to learn how to use hyper! <3

I had some thoughts on adjusting slightly where this content goes, and then I think we'd be ready to go. Thank you again!

**Comment by n1haldev** [comment]
I have implemented the necessary changes.

**Review by seanmonstar — APPROVED** [review]
Beautiful, thank you!

**Pull request merged**